### PR TITLE
uopz override functions

### LINKED ISSUE #7: HTTP calls via file_get_contents() not captured
Found this issue during investigation of a kit and noticed it making telegram calls without it being intercepted by mitmproxy.

PoC:
```php
<?php
function telegram($msg) {
    $url='https://api.telegram.org/';
    $options=array(
        'http'=>array(
            'method'=>'POST',
            'header'=>"Content-Type:application/x-www-form-urlencoded\r\n",
            'content'=>$msg
        )
    );

    $context=stream_context_create($options);
    
    $result=file_get_contents($url,false,$context);
    
    return $result;
}

function telegram_curl($msg) {
    $ch = curl_init();
    curl_setopt($ch, CURLOPT_URL,"https://api.telegram.org/");
    curl_setopt($ch, CURLOPT_POST, 1);
    curl_setopt($ch, CURLOPT_POSTFIELDS, "message=$msg");
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    $server_output = curl_exec($ch);
    curl_close ($ch);
    return $server_output;
}

$message1 = "stream!";
$message2 = "curl!";

telegram($message);
telegram_curl($message);
?>
```

Running the above will show only the calls being made via curl being intercepted by mitmproxy.

`file_get_contents` does not respect proxy config unless declared within the options array, there are likely other functions which will have the same behaviour.

### PR DESCRIPTION
**Pull request by zmallen**
Fixes #7 . Lots of files here, but basically use uopz to monkeypatch internal functions and force use of mitmproxy.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -32,8 +32,7 @@ services:
     build:
       context: ./docker/mitmproxy
     container_name: 'mitmproxy'
-    command: bash -c "mitmweb --web-host 0.0.0.0 --set confdir=/home/mitmproxy/.mitmproxy" 
-
+    command: bash -c "mitmweb --web-host 0.0.0.0 --set confdir=/home/mitmproxy/.mitmproxy --set relax_http_form_validation"
```
Comment (zmallen) [inline docker-compose.yml:35]: need to remove

File: configs/web/vhosts/default.conf [hunk 2]
```diff
@@ -14,4 +14,8 @@
 	DumpIOInput On
 	DumpIOOutput On
 	LogLevel dumpio:trace7
+#  	SetEnv http_proxy http://mitmproxy:8080
```
Comment (zmallen) [inline configs/web/vhosts/default.conf:17]: need to remove commented out directives

File: docker/web/Dockerfile [hunk 3]
```diff
@@ -58,11 +63,13 @@ RUN mkdir -p /usr/local/ioncube
 COPY ioncube_loader_lin_7.4.so /usr/local/ioncube
 
 # Enable apache modules
+RUN a2enmod proxy_http
```
Comment (zmallen) [inline docker/web/Dockerfile:66]: dont need

File: docker/web/auto_prepend.php [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<?php
+/*
+ * rename file_get_contents to an internal name
+ * add a new reserved file_get_contents
+ * within the new one, check if the first argument is http/https, if it is write to log, if it isnt then call int_file_get_contents
+ */
+runkit7_function_rename('file_get_contents', 'int_file_get_contents');
+// TODO: add logic in separate file perhaps to load the correct code in, also make sure the arguments line up to pass to int_file_get_contents
+runkit7_function_add('file_get_contents, 'echo "hello there";');
```
Comment (zmallen) [inline docker/web/auto_prepend.php:9]: make the magic happen

**Comment by zmallen** [comment]
I forgot to stash the frontend, so I might as well finish that too in here

**Review by sysgoblin — APPROVED** [review]
looks bang tidy

**Pull request merged**